### ISSUE TITLE
Style/StringLiterals ignores interpolation in multi line strings

### ISSUE THREAD

**Issue by maia** [bbatsov/rubocop#2549]
Rubocop seems to check multi-lined strings one line by the other, and if there's no interpolation in the current line it will report. I do believe that all lines of a multi-line string should use double-quoted strings if there's an interpolation in at least one line:
````
$ cat test.rb
"This is a multi-line string using interpolation #{Time.now.utc} "\
"but rubocop thinks not all lines should use double quotes."

$ rubocop test.rb --only Style/StringLiterals
Inspecting 1 file
C

Offenses:

test.rb:2:1: C: Prefer single-quoted strings when you don't need string interpolation or special symbols.
"but rubocop thinks not all lines should use double quotes."
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
````
Is this an incorrect report by rubocop, or is it suggested to mix single and double quotes in multi-line strings?

**Comment by alexdowad** [comment]
This has already been brought up on this bug tracker before. The RC developers like the current style. If you would like to code up a PR to make this configurable, it will be appreciated.

**Comment by maia** [comment]
Thanks. I've searched the issues for StringLiterals and didn't find anything related to multi-line strings. Can you point me to the issue? I'd like to read the pro's for the current style, as I didn't even think that mixing quotes in a single string is even allowed:
````
"This #{Time.now}"\
'feels very wrong'
````
As for your suggestion, I've just taken a quick look at the code and don't think I could add a configuration option, as it seems as if the checked node is not the entire multi-lined string but each single line is checked one by one. Unfortunately I don't have enough knowledge to alter that behavior.

**Comment by alexdowad** [comment]
@maia It may have been in the [style guide repo](https://github.com/bbatsov/ruby-style-guide).

When you have string literals which are concatenated like this:

```ruby
"A string" \
'another one'
```

...they are parsed as a `dstr` node with one `str` node under it for each line.

So you could add an `on_dstr` hook to `StringLiterals` to make it catch those nodes. Check if all the children are `str` nodes, and if not, just return. Otherwise, if they are all `str`, then process them and add offenses as is appropriate.

Then, call `ignore_node(node)` on each of the `str` nodes. The `on_str` hook (which is in `StringHelp`) already skips nodes which are marked as "ignored", so you can avoid double-processing of the nodes that way.

**Comment by maia** [comment]
@alexdowad Thanks, I didn't find it in the style guide repo, so maybe the current style is not considered best practice. As for the PR, in case I can find a few spare hours to dig deeper in the internals of rubocop, I will make an attempt, but please don't count on it.

**Comment by alexdowad** [comment]
I think we can add a config parameter to enforce the use of consistent quote types for multiline-continuation strings. Question: what should the new config parameter be called?

**Comment by maia** [comment]
How about TreatMultiLinesAsGroup or ConsistentQuotesInMultiLines ?

**Comment by jonas054** [comment]
`ConsistentQuotesInMultilines` sounds good. Note that we treat multiline as a single word in configuration, e.g. `Style/MultilineMethodCallIndentation`.

**Comment by alexdowad** [comment]
Just pushed a fix for this one to my open PR. The new config parameter is called `ConsistentQuotesInMultiline`, as suggested.

**Comment by maia** [comment]
That's great news, thanks a lot!

**Comment by maia** [comment]
Thanks a lot!
